File: app/src/main/java/com/dev/lokeshkalal/creditCard/identifier/DinerCardIdentifier.java
package com.dev.lokeshkalal.creditCard.identifier;


import com.dev.lokeshkalal.creditCard.CreditCardType;

public class DinerCardIdentifier implements Identifier {
    String pattern;
    final static int MAX_NUMBER = 14;
    final static int MIN_NUMBER = 14;

    public DinerCardIdentifier() {
        pattern = "^3(?:0[0-5]|[68][0-9])[0-9]{0,11}$";
    }

    @Override
    public CardIdentifierResult validateCard(String cardNumner) {
        if (cardNumner.matches(pattern)) {
            return new CardIdentifierResult(CreditCardType.DINER, MAX_NUMBER, MIN_NUMBER);
        } else {
            return CardIdentifierResult.InvalidCard();
        }
    }
}

File: app/src/main/java/com/dev/lokeshkalal/creditCard/identifier/DinerCardIdentifier.kt
package com.dev.lokeshkalal.creditCard.identifier


import com.dev.lokeshkalal.creditCard.addCreditCard.CreditCardType

class DinerCardIdentifier : Identifier {
    internal var pattern: String

    init {
        pattern = "^36[0-9]{0,12}$"
    }

    override fun validateCard(cardNumner: String): CardIdentifierResult {
        return if (cardNumner.matches(pattern.toRegex())) {
            CardIdentifierResult(CreditCardType.DINER, MAX_NUMBER, MIN_NUMBER)
        } else {
            CardIdentifierResult.InvalidCard()
        }
    }

    companion object {
        internal val MAX_NUMBER = 14
        internal val MIN_NUMBER = 14
    }
}
